Makefile

```diff
@@ -10,21 +10,31 @@ MOCKGEN := $(GOPATH)/bin/mockgen
 $(MOCKGEN):
 	go get -u github.com/golang/mock/mockgen
 
+cleanup:
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js/*
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js-no-definitions/*
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js-deprecated/*
+	rm -rf $(GOPATH)/src/$(PKG)/samples/gen-js-wag-patch/*
+	rm -rf $(PKG)/samples/gen-go/*
+	rm -rf $(PKG)/samples/gen-go-no-definitions/*
+	rm -rf $(PKG)/samples/gen-go-deprecated/*
+	rm -rf $(PKG)/samples/gen-go-wag-patch/*
```

Comment:
yes, the Go generation nukes the pre-existing directory

i think everyone is going to want that to be the default, and if we force users to do it people are going to be confused. i understand the concern, but i think i would prefer encouraging people not to change the code in those packages.